### PR TITLE
e2e image test

### PR DESCRIPTION
**Pull request by Crazykev**
1. ImageList will return digest only image.
2. make image pulling timeout related with pulling progress, if it stuck(without progress) over 1 minute, context will be cancelled.
3. add e2e image test cases

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/frakti/26)

### CODE REVIEW

File: test/e2e/framework/util.go [hunk 1]
```diff
@@ -90,3 +90,28 @@ func NewUUID() string {
...
 }
+
+func ClearAllImages(client *FraktiClient) {
+	imageList, err := client.ListImages(nil)
+	ExpectNoError(err, "Failed to get image list: %v", err)
+	for _, image := range imageList {
+		if len(image.RepoTags) == 0 {
+			for _, rd := range image.RepoDigests {
+				err = client.RemoveImage(&kubeapi.ImageSpec{
+					Image: &rd,
```
Comment (feiskyer) [inline test/e2e/framework/util.go]: nit: use a local variable instead

File: test/e2e/framework/util.go [hunk 2]
```diff
@@ -90,3 +90,28 @@ func NewUUID() string {
...
+				})
+				ExpectNoError(err, "Failed to remove image: %v", err)
+			}
+			continue
+		}
+		for _, rt := range image.RepoTags {
+			err = client.RemoveImage(&kubeapi.ImageSpec{
+				Image: &rt,
```
Comment (feiskyer) [inline test/e2e/framework/util.go]: Same with above

File: pkg/hyper/client.go [hunk 3]
```diff
@@ -235,34 +239,70 @@ func (c *Client) PullImage(image, tag string, auth *types.AuthConfig, out io.Wri
...
-			n, err := out.Write(res.Data)
+	for {
+		select {
+		case <-ticker.C:
+			//pulling image timeout, cancel it
+			cancel()
+			return fmt.Errorf("Cancel pulling image %q because of no progress for %v", image, defaultImagePullingStuckTimeout)
+		case err = <-errC:
 			if err != nil {
+				cancel()
```
Comment (feiskyer) [inline pkg/hyper/client.go]: Same with above.

File: pkg/hyper/client.go [hunk 4]
```diff
@@ -235,34 +239,70 @@ func (c *Client) PullImage(image, tag string, auth *types.AuthConfig, out io.Wri
...
 		}
+	}()
 
-		if out != nil {
-			n, err := out.Write(res.Data)
+	for {
+		select {
+		case <-ticker.C:
+			//pulling image timeout, cancel it
+			cancel()
```
Comment (feiskyer) [inline pkg/hyper/client.go]: Why added `cancel()` here? Since it is already called by `defer`.

Comment (Crazykev) [inline pkg/hyper/client.go]: yep, trivial cancel, we don't need this

File: test/e2e/framework/util.go (hunk 1 above)
Comment (Crazykev) [inline test/e2e/framework/util.go]: Not sure I got the point here. 
```go
var string repoDigest
for _, repoDigest = range image.RepoDigests {
 				err = client.RemoveImage(&kubeapi.ImageSpec{
 					Image: &repoDigest,
```
Like this one?

Comment (feiskyer) [inline test/e2e/framework/util.go]: No,  it is not recommended to take the address of range variables because Golang reused it for whole loop.

**Comment by Crazykev** [comment]
@feiskyer  Updated.

**Comment by feiskyer** [comment]
LGTM.

**Pull request merged**